### ISSUE TITLE
[Feature Request] Add Objects365 open dataset

### ISSUE THREAD

**Issue by hglee98** [Nota-NetsPresso/netspresso-trainer#481]
### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues.

### Is there any feature that you would like to add?

To improve AP on COCO validation set for RT-DETR, we should add support for the Objects365 dataset in netspresso-trainer. This aligns with the official RT-DETR implementation's recommendation to use pretrained weights from Objects365 to train and validate on COCO dataset.

### Is there any suggestion (or solution) to solve this issue?

- Implement data downloading and preprocessing for Objects365
- Update documentation to include Objects365 usage
- Test and validate performance improvements

### Additional context.

Related: #480 

### Will you help us by submitting a PR?

- [X] I will submit a PR about this issue.
